src/StackExchange.Redis/RedisDatabase.cs

```diff
@@ -721,6 +721,42 @@ public Task<bool> KeyExpireAsync(RedisKey key, DateTime? expiry, CommandFlags fl
             return ExecuteAsync(msg, ResultProcessor.Boolean, server: server);
         }
 
+        public bool KeyExpire(RedisKey key, TimeSpan expiryTime, ExpiryOption expiryOption, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = GetExpiryMessage(key, flags, expiryTime, out ServerEndPoint? server, expiryOption);
+            return ExecuteSync(msg, ResultProcessor.Boolean, server: server);
+        }
```

Comment:
Agree

Reply:
@NickCraver One small problem here that I'm not sure how to fix:
I wanted to make the TimeSpan and DateTime not nullable in these new signatures to avoid using PERSIST, but then I can't call the new methods from the existing methods.
In this current implementation, the user can use `null` as the `expiry` parameter and `ExpiryWhen.Always` as `when` and get PERSIST command which is a bit weird.

Do you think it's ok? Otherwise, we can make it not nullable again.

Reply:
I tweaked the approach here in latest - think current is good to go :+1: